### PR TITLE
ENT-3963 Fix self upgrade on AIX

### PR DESCRIPTION
**Pull request by nickanderson**

### CODE REVIEW

File: cfe_internal/core/watchdog/watchdog.cf [hunk 1]
```diff
@@ -17,18 +17,99 @@ bundle agent cfe_internal_core_watchdog(state)
...
-      "WARNING $(this.bundle): Currently only supports /etc/cron.d on systems that have pgrep in the the stdlib paths bundle.";
+    !(use_cfe_internal_core_watchdog_cron_d|aix)::
+      "WARNING $(this.bundle): Currently only supports /etc/cron.d on systems that have pgrep in the the stdlib paths bundle and AIX hosts.";
+}
+
+bundle agent cfe_internal_core_watchdog_aix(state)
+# @brief Manage watchdog state on aix
+# @param state enabled|disabled
+# - When enabled a cron job will be present to start cf-execd if it's not running.
+# - When disabled cron jobs including the the text CFEngine watchdog will not be present.
```
Comment (vpodzime) [inline cfe_internal/core/watchdog/watchdog.cf]: Double *the*

File: cfe_internal/core/watchdog/watchdog.cf [hunk 2]
```diff
@@ -17,18 +17,99 @@ bundle agent cfe_internal_core_watchdog(state)
...
+        create => "true";
+
+      # The watchdog script takes care of detecting conditions and launching
+      # necessary components.
+
+      "$(sys.bindir)/watchdog"
+        create => "true",
+        template_method => "mustache",
+        perms => mog( "700", "root", "system" ),
+        edit_template => "$(this.promise_dirname)/templates/watchdog.mustache";
```
Comment (vpodzime) [inline cfe_internal/core/watchdog/watchdog.cf]: Where does this template come from?

Comment (vpodzime) [reply cfe_internal/core/watchdog/watchdog.cf]: I can see it now in the last commit. I think it should be squashed into this commit.

**Comment by cf-bottom** [comment]
Thanks for submitting a PR! Maybe @craigcomstock can review this?

File: cfe_internal/core/watchdog/watchdog.cf (hunk 1 above)
Comment (craigcomstock) [reply cfe_internal/core/watchdog/watchdog.cf]: Something more like "When disabled cron jobs including the CFEngine watchdog will not be present."

File: update.cf [hunk 3]
```diff
@@ -68,7 +68,7 @@ bundle agent cfengine_internal_standalone_self_upgrade
         string => "$(sys.statedir)/MPF/desired-cfengine-package-version.json";
 
       "desired_pkg_data" -> { "ENT-3592" }
-        data => readjson( $(desired_pkg_data_path) ),
+        data => readjson( $(desired_pkg_data_path), inf ),
         if => fileexists( $(desired_pkg_data_path) );
```
Comment (craigcomstock) [inline update.cf]: does "inf" translate to "infinity" somehow? I assume so but don't know where to look to check. Is it documented somewhere on docs.cfengine.com?

Comment (craigcomstock) [reply update.cf]: yep. didn't scroll down enough. got it!

File: cfe_internal/core/watchdog/watchdog.cf (hunk 1 above)
Comment (nickanderson) [reply cfe_internal/core/watchdog/watchdog.cf]: Thank you.

File: cfe_internal/core/watchdog/watchdog.cf (hunk 2 above)
Comment (nickanderson) [reply cfe_internal/core/watchdog/watchdog.cf]: Yes, i will squash them. thanks.

**Pull request merged**